aten/src/ATen/native/cuda/Dropout.cu

```diff
@@ -92,12 +100,26 @@ fused_dropout_kernel_vec(at::cuda::detail::TensorInfo<const scalar_t, IndexType>
       gridxvec_loop_state ^= 1;
     }
 
+#ifdef USE_ROCM
+    if (VEC == 8) {
+      rand1 = curand_uniform4(&state);
+    }
+#endif
+
     rand.x = rand.x < p;
     rand.y = rand.y < p;
-    if (VEC == 4) {
+    if (VEC >= 4) {
       rand.z = rand.z < p;
       rand.w = rand.w < p;
     }
+#ifdef USE_ROCM
+    if (VEC == 8) {
+      rand1.x = rand1.x < p;
```

Comment:
NIT,
Better to name
rand -> rand1 
rand1 -> rand2
so that all 8 VEC are stored.

Reply:
I didn't want to do that for better PR readability.
Can do that in a follow-up PR, if needed